### PR TITLE
Add a property to configure the max error retry on the ssm client use…

### PR DESCRIPTION
**Pull request by fredboutin**
…d in the property source.

### CODE REVIEW

File: src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceConfigurationProperties.java [hunk 1]
```diff
@@ -19,8 +20,9 @@
                                                                        "signingRegion");
     public static final String MULTI_REGION_SSM_CLIENT_REGIONS = joinWithDot(SOURCE_PREFIX,
                                                                              "multiRegion",
-                                                                             "ssmClient",
+                                                                             SSM_CLIENT,
                                                                              "regions");
+    public static final String MAX_ERROR_RETRY = joinWithDot(SOURCE_PREFIX, SSM_CLIENT, "maxErrorRetry");
```
Comment (jebeaudet) [inline src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceConfigurationProperties.java:25]: `maxErrorRetries` like in the getter?

File: src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceConfigurationStrategy.java [hunk 2]
```diff
@@ -35,23 +35,29 @@ public void configureParameterStorePropertySources(ConfigurableEnvironment envir
...
                .skip(1)
                .forEach(region -> environment.getPropertySources()
-                                             .addFirst(buildParameterStorePropertySource(region, false)));
+                                             .addFirst(buildParameterStorePropertySource(ssmClientBuilder,
+                                                                                         region,
+                                                                                         false)));
     }
 
-    private ParameterStorePropertySource buildParameterStorePropertySource(String region, boolean haltBoot)
+    private ParameterStorePropertySource buildParameterStorePropertySource(AWSSimpleSystemsManagementClientBuilder ssmClientBuilder,
```
Comment (jebeaudet) [inline src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceConfigurationStrategy.java:49]: all those formatting changes are normal?

Comment (fredboutin) [reply src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceConfigurationStrategy.java:49]: yup

File: src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceConfigurationProperties.java (hunk 1 above)
Comment (fredboutin) [reply src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceConfigurationProperties.java:25]: the aws parameter is maxErrorRetry, so I kept the same naming

File: src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceConfigurationStrategy.java (hunk 2 above)
Comment (fredboutin) [reply src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceConfigurationStrategy.java:49]: dunno why github does not highlight the change but there is a parameter added

**Pull request merged**